### PR TITLE
Keyword feature

### PR DESCRIPTION
**Pull request by gyosida**
There are two things missing that will be addressed in separate tickets: Saving the state of the view is missing and an issue when returning to this tab after creating a star, the list is not updated with the possible new keyword.

### CODE REVIEW

**Comment by gyosida** [comment]
@PedroCarrillo  @cpinan

File: app/build.gradle [hunk 1]
```diff
@@ -49,4 +49,5 @@ dependencies {
     compile 'com.github.bumptech.glide:glide:3.7.0'
     compile 'com.aurelhubert:ahbottomnavigation:1.1.8'
     compile 'com.squareup.picasso:picasso:2.5.2'
+    compile 'com.android.support:support-v4:23.3.0'
```
Comment (PedroCarrillo) [inline app/build.gradle]:
```
compile "com.android.support:support-v4:$supportLibrariesVersion"
```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -38,6 +38,9 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity
+            android:name=".ui.stars.keyword.KeywordsActivity"
+            android:label="@string/title_keywords_activity"/>
```
Comment (PedroCarrillo) [inline app/src/main/AndroidManifest.xml:46]: you are literally adding and space to the other strings but the one you add it doesn't have any space at the end?

File: app/src/main/java/com/belatrixsf/allstars/networking/retrofit/api/ServerPaths.java [hunk 3]
```diff
@@ -40,17 +40,22 @@
     String PAGE = "page";
 
     String QUERY_PAGE = "page";
+    String QUERY_SEARCH = "search";
+    String QUERY_NEXT = "next";
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/allstars/networking/retrofit/api/ServerPaths.java:44]: not used

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (gyosida) [inline app/src/main/AndroidManifest.xml:46]: are you blind?

File: app/src/main/java/com/belatrixsf/allstars/networking/retrofit/api/ServerPaths.java [hunk 4]
```diff
@@ -40,17 +40,22 @@
...
 
     String EMPLOYEE_AUTHENTICATE = EMPLOYEE + "/authenticate/";
     String EMPLOYEE_LIST = EMPLOYEE + "/list/";
     String EMPLOYEE_BY_ID = EMPLOYEE + "/{" + EMPLOYEE_ID + "}";
     String EMPLOYEE_CATEGORIES = EMPLOYEE + "/{" + EMPLOYEE_ID + "}/category/list";
     String EMPLOYEE_SUBCATEGORY_LIST = STAR + "/{" + EMPLOYEE_ID + "}" + "/subcategory/list";
+    String EMPLOYEE_SEARCH_TERM = EMPLOYEE + "/list";
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/allstars/networking/retrofit/api/ServerPaths.java]: Same as EMPLOYEE_LIST

Comment (gyosida) [inline app/src/main/java/com/belatrixsf/allstars/networking/retrofit/api/ServerPaths.java]: good catch

File: app/src/main/java/com/belatrixsf/allstars/ui/common/views/searchingview/SearchingView.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+        clearImageButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                searchingEditText.setText("");
+            }
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/allstars/ui/common/views/searchingview/SearchingView.java:95]: smart view

File: app/src/main/java/com/belatrixsf/allstars/services/EmployeeServerService.java [hunk 6]
```diff
@@ -55,6 +55,16 @@ public void getEmployee(int employeeId, AllStarsCallback<Employee> callback) {
     }
 
     @Override
+    public void getEmployees(AllStarsCallback<SearchEmployeeResponse> callback) {
+        employeeAPI.getEmployees(null).enqueue(new RetrofitCallback<SearchEmployeeResponse>(callback));
+    }
+
+    @Override
+    public void getEmployees(String searchTerm, AllStarsCallback<SearchEmployeeResponse> callback) {
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/allstars/services/EmployeeServerService.java]: getEmployeeSearchList works for both cases

File: app/src/main/java/com/belatrixsf/allstars/services/EmployeeService.java [hunk 7]
```diff
@@ -35,6 +35,8 @@
 
     void authenticate(String username, String password, AllStarsCallback<AuthenticationResponse> callback);
     void getEmployee(int employeeId, AllStarsCallback<Employee> callback);
+    void getEmployees(AllStarsCallback<SearchEmployeeResponse> callback);
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/allstars/services/EmployeeService.java]: Same here

**Pull request merged**